ddl/partition.go

```diff
@@ -1690,42 +1690,47 @@ func checkPartitionExprFuncAllowed(_ sessionctx.Context, _ *model.TableInfo, e a
 }
 
 func checkPartitionExprArgs(_ sessionctx.Context, tblInfo *model.TableInfo, e ast.ExprNode) error {
-	expr, ok := e.(*ast.FuncCallExpr)
-	if !ok {
-		return nil
-	}
-	switch expr.FnName.L {
-	case ast.ToDays, ast.ToSeconds, ast.DayOfMonth, ast.Month, ast.DayOfYear, ast.Quarter, ast.YearWeek,
-		ast.Year, ast.Weekday, ast.DayOfWeek, ast.Day:
-		return errors.Trace(checkResultOK(hasDateArgs(tblInfo, expr)))
-	case ast.Hour, ast.Minute, ast.Second, ast.TimeToSec, ast.MicroSecond:
-		return errors.Trace(checkResultOK(hasTimeArgs(tblInfo, expr)))
-	case ast.UnixTimestamp:
-		return errors.Trace(checkResultOK(hasTimestampArgs(tblInfo, expr)))
-	case ast.FromDays:
-		if err := checkResultOK(hasDateArgs(tblInfo, expr)); err != nil {
-			return errors.Trace(err)
-		}
-		return errors.Trace(checkResultOK(hasTimeArgs(tblInfo, expr)))
-	case ast.Extract:
-		switch expr.Args[0].(*ast.TimeUnitExpr).Unit {
-		case ast.TimeUnitYear, ast.TimeUnitYearMonth, ast.TimeUnitQuarter, ast.TimeUnitMonth, ast.TimeUnitDay:
+	switch expr := e.(type) {
```

Comment:
do not change here, this function is used to the `expression args`. you should put `opcode.Not2` and `opcode.Not` to function `checkPartitionExprAllowed`

Reply:
> do not change here, this function is used to check the `function's` args. you should put `opcode.Not2` and `opcode.Not` to function `checkPartitionExprAllowed`

@xiongjiwei Update. PTAL~

Reply:
we can combine `checkPartitionExprFuncAllowed` and `checkPartitionExprAllowed`. I forgot why I split it into two functions. And make `allowedFuncMap` a global variable in this file to avoid memory allocation every time.